cloud/scope/azurestackedgeippool.go

```diff
@@ -36,104 +36,104 @@ var lock = &sync.Mutex{}
+func (s *AzureStackEdgeIPPoolScope) PatchObject(ctx context.Context) error {
 	return s.patchHelper.Patch(
 		ctx,
-		s.AzureIPPool,
+		s.AzureStackEdgeIPPool,
 		patch.WithOwnedConditions{Conditions: []clusterv1.ConditionType{
 			clusterv1.ReadyCondition,
 			infrav1.NetworkInfrastructureReadyCondition,
 		}})
 }
 
 // Close closes the current scope persisting the cluster configuration and status.
-func (s *AzureIPPoolScope) Close(ctx context.Context) error {
-	return s.patchHelper.Patch(ctx, s.AzureIPPool)
+func (s *AzureStackEdgeIPPoolScope) Close(ctx context.Context) error {
+	return s.patchHelper.Patch(ctx, s.AzureStackEdgeIPPool)
 }
 
-func (s *AzureIPPoolScope) GetIPPoolObj(ctx context.Context, namespace string, ipPoolName string) (*infrav1.AzureIPPool, error) {
-	azureIPPoolName := client.ObjectKey{
+func (s *AzureStackEdgeIPPoolScope) GetIPPoolObj(ctx context.Context, namespace string, ipPoolName string) (*infrav1.AzureStackEdgeIPPool, error) {
+	azureStackEdgeIPPoolName := client.ObjectKey{
 		Namespace: namespace,
 		Name:      ipPoolName,
 	}
-	azureippool := &infrav1.AzureIPPool{}
-	if err := s.Client.Get(ctx, azureIPPoolName, azureippool); err != nil {
-	return azureippool, errors.Wrapf(err, "Failed to retrieve Azure IP pool")
+	azurestackedgeippool := &infrav1.AzureStackEdgeIPPool{}
+	if err := s.Client.Get(ctx, azureStackEdgeIPPoolName, azurestackedgeippool); err != nil {
+	return azurestackedgeippool, errors.Wrapf(err, "Failed to retrieve Azure IP pool")
 	}
-	return azureippool, nil
+	return azurestackedgeippool, nil
 }
 
 /*Reconcile IP pool gets the IP from IP pool, adds it to allocated IP pool and updates the IP pool. Machine IP is "" for worker VMs and
 equal to APIServerIP for control plane VM*/
-func (s *AzureIPPoolScope) ReconcileIPs(ctx context.Context, machineScope *MachineScope, ippoolname string, machineIP string) error {
+func (s *AzureStackEdgeIPPoolScope) ReconcileIPs(ctx context.Context, machineScope *MachineScope, ippoolname string, machineIP string) error {
 	lock.Lock()
 	machineScope.Info(fmt.Sprintf("Locking reconcile IP for machine %s", machineScope.Name()))
 	defer func() {
 		machineScope.Info(fmt.Sprintf("Unlocking reconcile IP for machine %s", machineScope.Name()))
-     lock.Unlock()   
+    lock.Unlock()   
     }()
 	
-	azureIPPool, err := s.GetIPPoolObj(ctx, "default", ippoolname)
+	azureStackEdgeIPPool, err := s.GetIPPoolObj(ctx, "default", ippoolname)
     if err!=nil {
         return errors.Wrapf(err, "Failed to retrieve Azure IP pool")
     }
 	
-	if azureIPPool.Spec.IPPools == nil {
+	if azureStackEdgeIPPool.Spec.IPPools == nil {
 		return errors.New("IP Pool cannot be nil")
 	}
 
-	for index, _ := range(azureIPPool.Spec.IPPools) {
-		ippool := azureIPPool.Spec.IPPools[index]        
+	for index, _ := range(azureStackEdgeIPPool.Spec.IPPools) {
+		ippool := azureStackEdgeIPPool.Spec.IPPools[index]        
 		networkInterface := BuildNetworkInterfaceSpec(machineScope, ippool, ippool.Name)
```

Comment:
Should we move AcceleratedNetworking to yaml file instead of assuming all vnet's beside mgmt has AcceleratedNetworking enabled

Reply:
Done in the latest commit.  https://github.com/niachary/cluster-api-provider-azure/blob/b521e26ee8a14e47ef9708951321a7ee207d9147/cloud/scope/azurestackedgeippool.go#L249
Reading the acceleratedNetworking value from the input and passing it down to network interface create call.